config/config.go

```diff
@@ -158,22 +162,90 @@ var (
 
 // Config is the top-level configuration for Prometheus's config files.
 type Config struct {
-	GlobalConfig   GlobalConfig    `yaml:"global"`
-	AlertingConfig AlertingConfig  `yaml:"alerting,omitempty"`
-	RuleFiles      []string        `yaml:"rule_files,omitempty"`
-	ScrapeConfigs  []*ScrapeConfig `yaml:"scrape_configs,omitempty"`
+	GlobalConfig      GlobalConfig    `yaml:"global"`
+	AlertingConfig    AlertingConfig  `yaml:"alerting,omitempty"`
+	RuleFiles         []string        `yaml:"rule_files,omitempty"`
+	ScrapeConfigs     []*ScrapeConfig `yaml:"scrape_configs,omitempty"`
+	ScrapeConfigFiles []string        `yaml:"scrape_config_files,omitempty"`
 
 	RemoteWriteConfigs []*RemoteWriteConfig `yaml:"remote_write,omitempty"`
 	RemoteReadConfigs  []*RemoteReadConfig  `yaml:"remote_read,omitempty"`
 }
 
+// ScrapeConfigFile is the configuration for scrape config files.
+type ScrapeConfigFile struct {
+	ScrapeConfigs []*ScrapeConfig `yaml:"scrape_configs"`
+}
+
+// LoadScrapeConfigFiles parses the given YAML files from scrape_config_files
+// and adds ScrapeConfigs from all of them.
+func (c *Config) LoadScrapeConfigFiles() error {
+	var scrapeConfigFiles []string
+	for _, pat := range c.ScrapeConfigFiles {
+		fs, err := filepath.Glob(pat)
+		if err != nil {
+			// The only error can be a bad pattern.
+			return errors.Wrapf(err, "error retrieving scrape config files for %s", pat)
+		}
+		// There is no scrape config file that matches pattern.
+		if len(fs) == 0 {
+			return errors.Errorf("no scrape config files found for %s", pat)
```

Comment:
We should not error here; as the path is a glob, there could be no files.

Reply:
My line of thought was `filepath.Glob` returns err only when pattern is bad, If they provide a valid pattern and there are no files then it probably means there was a misspell or typo. What would be the ideal behaviour here? Should we err out here or skip such cases?

Reply:
This should be silently ignored if there  are no files, to match what we do for rule files.